handlers/ocr-handler/server.py

```diff
@@ -0,0 +1,149 @@
+    except jsonschema.exceptions.ValidationError as error:
+        logging.error(error)
+        return jsonify("Invalid Request JSON format"), 400
+    
+    # Check preprocessor data
+    preprocessors = content['preprocessors']
+    
+    # No OCR preprocessor
+    if 'ca.mcgill.a11y.image.preprocessor.ocr' not in preprocessors:
+        logging.error("No OCR preprocessor found")
+        response = {
+            "request_uuid": content["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+    
+    ocr_data = preprocessors['ca.mcgill.a11y.image.preprocessor.ocr']
+    
+    # OCR lines empty
+    if len(ocr_data['lines']) == 0:
+        logging.error("OCR lines empty")
+        response = {
+            "request_uuid": content["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+
+    # Text renderer not supported
+    if 'ca.mcgill.a11y.image.renderer.Text' not in content['renderers']:
+        logging.error("Text renderer not supported")
+        response = {
+            "request_uuid": content["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+    
+    # Get text renderer data
+    text = 'The following ' + str(len(ocr_data['lines'])) + ' lines of text were found in the image: '
+    for i, line in enumerate(ocr_data['lines']):
+        line_text = 'Start of line ' + str(i+1) + ': ' + line['text'] + '. End of line ' + str(i+1) + '. '
+        text += line_text
+    
+    response = {
+        "request_uuid": content["request_uuid"],
+        "timestamp": int(time.time()),
+        "renderings": [
+            {
+                "type_id": "ca.mcgill.a11y.image.renderer.Text",
+                "confidence": 50,
+                "description": "The text found in an image.",
```

Comment:
Same "image" question as above. Are there any preconditions on the OCR preprocessor (i.e., checks against the first categoriser)?

Reply:
No, not as of this iteration